contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,971 @@
+        }
+    }
+
+    /// @dev this function assumes there are no uncollected tokens in the clPool owned by the.
+    ///      strategy contract. For that reason any liquidity withdrawals also collect the tokens.
+    function _updateUnderlyingAssets() internal {
+        if (tokenId == 0) {
+            underlyingAssets = 0;
+            emit UnderlyingAssetsUpdated(underlyingAssets);
+            return;
+        }
+
+        uint128 _liquidity = _getLiquidity();
+
+        /**
+         * Our net value represent the smallest amount of tokens we are able to extract from the position
+         * given our liquidity.
+         *
+         * The least amount of tokens extraditable from the position is where the active trading price is
+         * at the ticker 0 meaning the pool is offering 1:1 trades between WETH & OETHb. At that moment the pool
+         * consists completely of OETHb and no WETH.
+         *
+         * The more swaps from WETH -> OETHb happen on the pool the more the price starts to move towards the -1
+         * ticker making OETHb (priced in WETH) more expensive.
+         * 
+         * An additional note: when liquidity is 0 then the helper returns 0 for both token amounts. And the 
+         * function set underlying assets to 0.
+         */
+        (uint256 _wethAmount, uint256 _oethbAmount) = helper
+            .getAmountsForLiquidity(
+                sqrtRatioX96TickClosestToParity, // sqrtRatioX96
+                sqrtRatioX96TickLower, // sqrtRatioAX96
+                sqrtRatioX96TickHigher, // sqrtRatioBX96
+                _liquidity
+            );
+
+        require(_wethAmount == 0, "Non zero wethAmount");
+        underlyingAssets = _oethbAmount;
+        emit UnderlyingAssetsUpdated(underlyingAssets);
+    }
+
+    /**
+     * @notice Withdraw an `amount` of assets from the platform and
+     *         send to the `_recipient`.
+     * @param _recipient  Address to which the asset should be sent
+     * @param _asset      WETH address
+     * @param _amount     Amount of WETH to withdraw
+     */
+    function withdraw(
+        address _recipient,
+        address _asset,
+        uint256 _amount
+    ) external override onlyVault nonReentrant {
+        require(_asset == WETH, "Unsupported asset");
+        require(_recipient == vaultAddress, "Only withdraw to vault allowed");
+
+        uint256 _wethBalance = IERC20(WETH).balanceOf(address(this));
+        if (_wethBalance < _amount) {
+            require(tokenId != 0, "No liquidity available");
+            uint256 _additionalWethRequired = _amount - _wethBalance;
+            (uint256 _wethInThePool, ) = getPositionPrincipal();
+
+            if (_wethInThePool < _additionalWethRequired) {
+                revert NotEnoughWethLiquidity(
+                    _wethInThePool,
+                    _additionalWethRequired
+                );
+            }
+
+            uint256 shareOfWethToRemove = Math.min(
```

Comment:
Note that this would also require changing `_removeLiquidity` to take an actual liquidity amount to remove instead of a bps amount. I don't think we want to make that change this late. But just flagging it

Reply:
Yeah, think we probably shouldn't change at this point, but it's a very clean solution.